plugin/src/main/java/com/h3xstream/findsecbugs/android/CaseInsensitivePackageNameComparisonDetector.java

```diff
@@ -40,7 +40,8 @@ public void sawOpcode(int seen){
 
 		if(seen == Const.INVOKEVIRTUAL &&
 		       (
-			   getNameConstantOperand().equals("getPackageName().equalsIgnoreCase")
+			   getNameConstantOperand().equals("getPackageName") &&
```

Comment:
This will not work.
Name Constant refer to the name of the method being called.
A method called can't be two  method at the same time.

This bug pattern is more complex to detect. Dataflow would need to be used.

I would recommend you to start with [easier rules or changes](https://github.com/find-sec-bugs/find-sec-bugs/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).

Reply:
Hi, thanks for reviewing my pull request. Can you at least guide me a little to fix my detector, please? Any tips appreciated.

Reply:
The easiest way to get started as a contributor is to pick as simpler task. Simplier tasks can get you familiar with the plugin lifecycle, tests, etc.

There a very small tweaks and additions that can be beneficial to the project. I have tagged few issues with the tag `[good first issue]`.
https://github.com/find-sec-bugs/find-sec-bugs/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22